src/components/SnippetCard.tsx

```diff
@@ -0,0 +1,125 @@
+import type React from 'react'
+import { useState } from 'react'
+import { Share2 } from 'lucide-react'
+
+import Upvote from 'assets/upvote.svg'
+import Upvoted from 'assets/upvoted.svg'
+import UpvoteHover from 'assets/upvote_hover.svg'
+import Star from 'assets/star.svg'
+import Starred from 'assets/starred.svg'
+import StarHover from 'assets/star_hover.svg'
+import { Button } from 'components/ui/button'
+import LiveblocksComments from 'components/LiveblocksComments'
+
+interface Snippet {
+  id: number;
+  radio_code: string;
+  name: string;
+  type: string;
+  channel: string;
+  state: string;
+  human_upvotes: number;
+}
+
+interface SnippetCardProps {
+  snippet: Snippet;
+  onSnippetClick: (id: number) => void;
+}
+
+const SnippetCard: React.FC<SnippetCardProps> = ({ snippet, onSnippetClick }) => {
+  const [isUpvoted, setIsUpvoted] = useState(false)
+  const [isHovered, setIsHovered] = useState(false)
+  const [isStarred, setIsStarred] = useState(false)
+  const [isStarHovered, setIsStarHovered] = useState(false)
+
+  const handleUpvoteClick = (e: React.MouseEvent) => {
+    e.stopPropagation()
+    setIsUpvoted(!isUpvoted)
+  }
```

Comment:
--- $${\color{green}[AI \space Generated \space Comment]}$$ ---
Consider using CSS hover effects instead of `useState` for hover states. This can improve performance by reducing unnecessary re-renders. For example:

```typescript
const SnippetCard: React.FC<SnippetCardProps> = ({ snippet, onSnippetClick }) => {
  const [isUpvoted, setIsUpvoted] = useState(false)
  const [isStarred, setIsStarred] = useState(false)

  // ... rest of the component
}
```

Then use CSS classes and `:hover` pseudo-classes to handle hover effects.